src/ParametricOptInterface.jl

```diff
@@ -406,6 +406,78 @@ function MOI.set(
     MOI.set(model.optimizer, MOI.ConstraintFunction(), c, f)
     return
 end
+function MOI.set(
+    model::Optimizer,
+    ::MOI.ConstraintFunction,
+    c::MOI.ConstraintIndex{MOI.ScalarAffineFunction{T},S},
+    f::F,
+) where {T,S<:MOI.AbstractSet,F}
+    if haskey(model.affine_added_cache, c)
+        MOI.set(
```

Comment:
this looks too simple, don't we have any other data cached for affine function in the `POI.Optimizer`?